pyanaconda/modules/payloads/payload/dnf/dnf.py

```diff
@@ -43,16 +47,40 @@ def type(self):
     @property
     def supported_source_types(self):
         """Get list of sources supported by DNF module."""
-        # TODO: Add supported sources when implemented
-        return None
+        return [
+            SourceType.CDROM,
+            SourceType.HDD,
+            SourceType.HMC,
+            SourceType.NFS,
+            SourceType.REPO_FILES,
+            SourceType.URL
+        ]
 
     def process_kickstart(self, data):
         """Process the kickstart data."""
-        pass
+        source_types = SourceFactory.get_type_for_kickstart(data)
+
+        if len(source_types) > 1:
+            types_strs = map(lambda x: "\"{}\"".format(x.ks_data.currentCmd), source_types)
+            msg = "Only one of {} payload sources can be used!".format(", ".join(types_strs))
+            lineno = source_types[-1].ks_data.lineno
+            raise KickstartParseError(msg, lineno=lineno)
+
+        if len(source_types) == 0:
+            return
+
+        source_module = SourceFactory.create_source(source_types[0].source_type)
+        source_module.process_kickstart(source_types[0].ks_data)
+
+        self.add_source(source_module)
+
+        # FIXME: This is a temporary workaround.
+        PayloadSourceContainer.to_object_path(source_module)
```

Comment:
This shouldn't be necessary if the `add_source` method changes a DBus property.

Reply:
I don't follow. How it should change the DBus property from the implementation class?

Reply:
Solved.